var/spack/repos/builtin/packages/modules/package.py

```diff
@@ -0,0 +1,86 @@
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+
+
+class Modules(Package):
+    """The Modules package provides for the dynamic modification of a user's
+    environment via modulefiles. This is a copy for the `Environment-Modules`
+    package, allowing users to experiment with different versions of `modules`
+    without breaking spack."""
+
+    homepage = "https://sourceforge.net/p/modules/"
+    url = "http://prdownloads.sourceforge.net/modules/modules-4.1.4.tar.gz"
+
+    version('4.1.4', sha256='965b6056ea6b72b87d9352d4c1db1d7a7f9f358b9408df2689d823b932d6aa7f')
+
+    variant('X', default=True, description='Build with X functionality')
+
+    # Dependencies:
+    depends_on('tcl', type=('build', 'link', 'run'))
+
+    def install(self, spec, prefix):
+        tcl_spec = spec['tcl']
+
+        # We are looking for tclConfig.sh
+        tcl_config_name = 'tclConfig.sh'
+        tcl_config_dir_options = [tcl_spec.prefix.lib,
+                                  tcl_spec.prefix.lib64]
+
+        tcl_config_found = False
+        for tcl_config_dir in tcl_config_dir_options:
+            tcl_config_found = os.path.exists(
+                join_path(tcl_config_dir, tcl_config_name))
+            if tcl_config_found:
+                break
+
+        if not tcl_config_found:
+            raise InstallError('Failed to locate ' + tcl_config_name)
+
+        cpp_flags = ['-DUSE_INTERP_ERRORLINE']
+
+        config_args = [
+            "--without-tclx",
+            "--with-tclx-ver=0.0",
+            "--prefix=" + prefix,
+            # It looks for tclConfig.sh
+            "--with-tcl=" + tcl_config_dir,
+            "--with-tcl-ver=%d.%d" % (
+                tcl_spec.version.version[0], tcl_spec.version.version[1]),
+            '--disable-debug',
+            '--disable-dependency-tracking',
+            '--disable-silent-rules',
+            '--disable-versioning',
+            '--datarootdir=' + prefix.share,
+            'CPPFLAGS=' + ' '.join(cpp_flags)
+        ]
+
+        if '~X' in spec:
+            config_args = ['--without-x'] + config_args
+
+        configure(*config_args)
+        make()
+        make('install')
```

Comment:
This seems to be a duplicate of the environment-modules or am I missing something?

Reply:
I'm sorry, this kinda slipped in unintentionally. I had this package in there to experiment with different modules versions in my machine.

If we downgrade the main one back to 3.2.10, we could use this one for fish support. This will make bootstrapping more complex though.